### ISSUE TITLE
Go get github.com/argoproj/argo-cd does not get the latest source code

### ISSUE THREAD

**Issue by samuel-sujith** [argoproj/argo-cd#4885]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [ yes] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ yes] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**
I am trying to create a new module which uses argo cd modules. My go.mod file goes like this
````yaml
module gitlab.internal.xxxxxx.com

go 1.14

require (
	github.com/argoproj/argo-cd v1.7.10
	github.com/argoproj/gitops-engine v0.1.3
	github.com/argoproj/pkg v0.3.0
	github.com/bombsimon/logrusr v1.0.0
	github.com/coreos/go-oidc v2.1.0+incompatible
	github.com/dgrijalva/jwt-go v3.2.0+incompatible
	github.com/ghodss/yaml v1.0.0
	github.com/go-kit/kit v0.9.0
	github.com/go-logr/logr v0.2.0
	github.com/gorilla/mux v1.8.0
	github.com/pkg/errors v0.9.1
	github.com/sirupsen/logrus v1.6.0
	github.com/skratchdot/open-golang v0.0.0-20160302144031-75fb7ed4208c
	github.com/spf13/cobra v1.1.1
	github.com/stretchr/testify v1.6.1
	golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
	golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
	google.golang.org/grpc v1.29.1
	gopkg.in/src-d/go-git.v4 v4.13.1
	k8s.io/client-go v11.0.1-0.20190816222228-6d55c1b1f1ca+incompatible
)

replace (
	k8s.io/api => k8s.io/api v0.19.2
	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.19.2
	k8s.io/apimachinery => k8s.io/apimachinery v0.19.2
	k8s.io/apiserver => k8s.io/apiserver v0.19.2
	k8s.io/cli-runtime => k8s.io/cli-runtime v0.19.2
	k8s.io/client-go => k8s.io/client-go v0.19.2
	k8s.io/cloud-provider => k8s.io/cloud-provider v0.19.2
	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.19.2
	k8s.io/code-generator => k8s.io/code-generator v0.19.2
	k8s.io/component-base => k8s.io/component-base v0.19.2
	k8s.io/cri-api => k8s.io/cri-api v0.19.2
	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.19.2
	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.19.2
	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.19.2
	k8s.io/kube-proxy => k8s.io/kube-proxy v0.19.2
	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.19.2
	k8s.io/kubectl => k8s.io/kubectl v0.19.2
	k8s.io/kubelet => k8s.io/kubelet v0.19.2
	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.19.2
	k8s.io/metrics => k8s.io/metrics v0.19.2
	k8s.io/node-api => k8s.io/node-api v0.19.2
	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.19.2
	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.19.2
	k8s.io/sample-controller => k8s.io/sample-controller v0.19.2
)
``````

Upon checking the vendor folder, some fields like EnableOCI in the types.go in the pkg.v1alpha1 folder is missing.

This causes problems in the compilation

**To Reproduce**

go mod vendor fetches incorrect argo-cd code

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue.

**Expected behavior**

**Screenshots**

**Version**

```shell
Paste the output from `argocd version` here.
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by jannfis** [comment]
Hey @samuel-sujith , as far as I can tell, you are pulling `github.com/argoproj/argo-cd v1.7.10` in your go.mod. 

The Helm OCI feature however is part of upcoming v1.8, and therefore not included in the v1.7.10 source code 

You could use a git commit SHA to a commit in the `release-1.8` branch for the argo-cd dependency until v1.8 is released.

**Comment by samuel-sujith** [comment]
Thanks a lot for reply. I will close this now.
